File: src/main/kotlin/app/xlui/pwitter/entity/Tweet.kt
package app.xlui.pwitter.entity

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "t_tweet")
class Tweet(
        @Id @GeneratedValue val id: Long = 0,
        val type: TweetType = TweetType.CONTENT,
        val content: String = "",

        @ManyToOne val user: User = User(),

        val createTime: LocalDateTime = LocalDateTime.now()
)

enum class TweetType {
    CONTENT,
    IMAGE,
    CONTENT_IMAGE,
    VIDEO
}

File: web/src/layouts/Tweet.js
import React from "react";

export default function (props) {
    const tweet = props.tweet
    console.log(tweet)
    return (
        <div>
            id: {tweet.id}
            <br/>
            content: {tweet.content}
            <br/>
            username: {tweet.user.username}
            <hr/>
        </div>
    )
}
